Marlin/src/gcode/probe/G30.cpp

```diff
@@ -56,12 +56,12 @@ void GcodeSuite::G30() {
   probe.use_probing_tool();
 
   // Convert the given logical position to native position
-  const xy_pos_t pos = {
-    parser.seenval('X') ? RAW_X_POSITION(parser.value_linear_units()) : current_position.x,
-    parser.seenval('Y') ? RAW_Y_POSITION(parser.value_linear_units()) : current_position.y
+  const xy_pos_t probepos = {
+    parser.seenval('X') ? RAW_X_POSITION(parser.value_linear_units()) : current_position.x + probe.offset.x,
+    parser.seenval('Y') ? RAW_Y_POSITION(parser.value_linear_units()) : current_position.y + probe.offset.y
```

Comment:
So we actually have an aesthetic choice here. We can either consider the current position (that the nozzle is over) as the position that we want to get a probed Z height at, or we can probe at the position near the nozzle that the probe is over. I suppose if the user wanted to probe under the nozzle position, we would have to move over to that position, do the probe, and then move the nozzle back, so that the next Z move can be done in consideration of the measured Z height.

It's a question of use-case. Not moving the nozzle over when no new XY is given seems the most intuitive, in that you could use `G1` to move the nozzle to a position, then do a `G30`, and you'll get no extra surprise lateral move. I think that probably best meets user expectations.

Anyway, without the above change, I guess we had the nozzle moving laterally on each `G30`, which is not what you would want. So this preferred fix —always considering the position as probe-relative— should prevent that weird behavior.

Meanwhile, I see that we have an extra Z move below…
```cpp
    if (raise_after == PROBE_PT_STOW)
      probe.move_z_after_probing();
```
…which I believe may no longer be needed because `probe.probe_at_point` should be doing that. However, since `probe.can_reach` could fail and prevent `probe.probe_at_point` from being called, it might be there to deal with that possibility.

Overall it's good that we're diving into these behaviors and getting them into more consistent shape.